sycl/test/extensions/annotated_usm/annotated_usm_negative.cpp

```diff
@@ -0,0 +1,87 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+
+// Expected failed tests for annotated USM allocation:
+// 1. Given properties contain invalid runtime property
+// 2. Given properties contain invalid compile-time property
+// 3. usm_kind in the property list conflicts with the function name
+// 4. required usm_kind is not provided in the property list
+
+#include <sycl/sycl.hpp>
+
+#include "fake_properties.hpp"
+
+#define TEST(f, args...)                                                       \
+  { auto ap = f(args); }
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+using alloc = sycl::usm::alloc;
+
+constexpr int N = 10;
+
+// clang-format off
+
+void testInvalidRuntimeProperty(sycl::queue &q) {
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid runtime property in the property list.}}
+  TEST((malloc_device_annotated<int>), N, q, properties{conduit, cache_config{large_slm}})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid runtime property in the property list.}}
+  TEST((malloc_shared_annotated<int>), N, q, properties{conduit, foo{foo_enum::b}})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid runtime property in the property list.}}
+  TEST((malloc_host_annotated<int>), N, q, properties{conduit, foz{0, 1}})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid runtime property in the property list.}}
+  TEST((malloc_annotated<int>), N, q, alloc::device, properties{conduit, foo{foo_enum::a}, foz{0, 1}})
+}
+
+// Expect error when the property list contain invalid comile-time property
+// Note that two errors are raised for each case, during:
+// 1. validating malloc input properties
+// 2. validating annotated_ptr properties
+void testInvalidCompileTimeProperty(sycl::queue &q) {
```

Comment:
Can you add some negative tests where the following invalid compile-time-constant properties are used:

sycl::ext::oneapi::experimental::work_group_size
sycl::ext::intel::experimental::ready_latency
sycl::ext::oneapi::experimental::device_image_scope

Reply:
test cases added